test/unit/storage/LockingResourceStore.test.ts

```diff
@@ -101,4 +105,63 @@ describe('A LockingResourceStore', (): void => {
     expect(lock.release).toHaveBeenCalledTimes(1);
     expect(order).toEqual([ 'acquire', 'release' ]);
   });
+
+  it('releases the lock on the resource when data has been read.', async(): Promise<void> => {
+    const readable = streamifyArray([ 1, 2, 3 ]);
+
+    source.getRepresentation = jest.fn(async(): Promise<any> =>
+      new Promise((resolve): any => delayedResolve(resolve, 'getRepresentation', {
+        data: readable,
+        dataType: DATA_TYPE_BINARY,
+        metadata: { raw: [], linkRel: { type: new Set() }, profiles: []} as RepresentationMetadata,
+      })));
+
+    const representation = await store.getRepresentation({ path: 'path' }, {});
+    const waitForMe = new Promise((resolve): any => {
```

Comment:
Would rename this to something like `drainData` so it's clear what we're waiting for.